pkg/metrics/collector/netlink_metrics.go

```diff
@@ -15,85 +15,287 @@ package collector
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"sync"
 	"syscall"
 
-	"github.com/GoogleCloudPlatform/netd/pkg/tcp_metrics/inetdiag"
-	"github.com/GoogleCloudPlatform/netd/pkg/tcp_metrics/parser"
-
+	"github.com/golang/glog"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/vishvananda/netlink/nl"
+	"github.com/vishvananda/netns"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/GoogleCloudPlatform/netd/pkg/tcp_metrics/inetdiag"
+	"github.com/GoogleCloudPlatform/netd/pkg/tcp_metrics/parser"
+	"github.com/GoogleCloudPlatform/netd/pkg/tcp_metrics/tcp"
 )
 
 var (
 	TCPConnectionsDesc = prometheus.NewDesc(
-		"tcp_connections",
+		"active_tcp_connections",
 		"Current amount of tcp connections on a node",
-		nil, nil,
+		[]string{"pod_name", "namespace_name"}, nil,
 	)
 
 	ActiveTCPRetransmits = prometheus.NewDesc(
 		"active_tcp_retransmits",
 		"Current amount of tcp retransmits on a node",
-		nil, nil,
+		[]string{"pod_name", "namespace_name"}, nil,
 	)
 )
 
+type netlinkCollector struct {
+}
+
+type netlinkStats struct {
+	retransmits    uint64
+	tcpConnections uint64
+}
+
+type podDef struct {
+	name      string
+	namespace string
+}
+
+var (
+	mapMux   sync.RWMutex
+	ipMap    = make(map[string]podDef)
+	nodeName = os.Getenv("CURRENT_NODE_NAME")
+)
+
+func init() {
+
+	glog.Infof("Initializing netlink metrics")
+	err := createPodWatch()
```

Comment:
Yup, my thoughts were to make it so this collector would only be enabled when we want fine grained metrics but since we also want the more aggregate metrics we can do this!